python/langsmith/evaluation/_runner.py

```diff
@@ -103,10 +105,52 @@ def evaluate(
     num_repetitions: int = 1,
     client: Optional[langsmith.Client] = None,
     blocking: bool = True,
-    experiment: Optional[Union[schemas.TracerSession, str, uuid.UUID]] = None,
+    experiment: Optional[EXPERIMENT_T] = None,
     upload_results: bool = True,
-) -> ExperimentResults:
-    r"""Evaluate a target system or function on a given dataset.
+    **kwargs: Any,
+) -> ExperimentResults: ...
+
+
+@overload
+def evaluate(
+    target: Union[Tuple[EXPERIMENT_T, EXPERIMENT_T]],
+    /,
+    data: DATA_T,
+    evaluators: Optional[Sequence[COMPARATIVE_EVALUATOR_T]] = None,
+    summary_evaluators: Optional[Sequence[SUMMARY_EVALUATOR_T]] = None,
+    metadata: Optional[dict] = None,
+    experiment_prefix: Optional[str] = None,
+    description: Optional[str] = None,
+    max_concurrency: Optional[int] = None,
+    num_repetitions: int = 1,
+    client: Optional[langsmith.Client] = None,
+    blocking: bool = True,
+    experiment: Optional[EXPERIMENT_T] = None,
+    upload_results: bool = True,
+    **kwargs: Any,
+) -> ComparativeExperimentResults: ...
+
+
+def evaluate(
+    target: Union[TARGET_T, Runnable, EXPERIMENT_T, Tuple[EXPERIMENT_T, EXPERIMENT_T]],
```

Comment:
If we're doing this, we should have _really_ good validation on what evaluators go with what targets. And we should fail early and loudly...

Reply:
Also debug logs saying which type of evaluation you're running up front